visualize.py

```diff
@@ -17,6 +17,7 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #
 
+from re import U
```

Comment:
I think this slipped in by accident, as it is never used. Could you remove this line?